analyzer/src/main/kotlin/managers/SpdxDocumentFile.kt

```diff
@@ -211,22 +211,32 @@ class SpdxDocumentFile(
             spdxDocument.packages
         }
 
-        val packages = nonProjectPackages.mapTo(sortedSetOf()) { pkg ->
+        fun PackageReference.getPackageReferenceIds(): List<Identifier> =
+            if (dependencies.isEmpty()) listOf(id) else dependencies.flatMap { it.getPackageReferenceIds() }
+
+        val packagesInScope = project.scopes.flatMap { it.dependencies }.flatMap { it.getPackageReferenceIds() }
+
+        val packages = nonProjectPackages.mapNotNullTo(sortedSetOf()) { pkg ->
```

Comment:
You might use the plain `map` function and get rid off the if-else construct in the lambda by applying a filter first:

```
val packages = nonProjectPackages.filter { packagesInScope.contains(it.toIdentifier() }
  .map { pgk ->
    ...
  }
```

Reply:
This also is not applicable anymore due to the changes done with this recommendation https://github.com/oss-review-toolkit/ort/pull/3647#discussion_r581189793